baselines/common/vec_env/subproc_vec_env.py

```diff
@@ -4,51 +4,62 @@
 from .vec_env import VecEnv, CloudpickleWrapper, clear_mpi_env_vars
 
 
-def worker(remote, parent_remote, env_fn_wrapper):
+def worker(remote, parent_remote, env_fn_wrappers):
     parent_remote.close()
-    env = env_fn_wrapper.x()
+    envs = [env_fn_wrapper() for env_fn_wrapper in env_fn_wrappers.x]
+    nenvs = len(envs)
     try:
         while True:
+            buf = [None for _ in range(nenvs)]
             cmd, data = remote.recv()
             if cmd == 'step':
-                ob, reward, done, info = env.step(data)
-                if done:
-                    ob = env.reset()
-                remote.send((ob, reward, done, info))
+                for i, env in enumerate(envs):
+                    buf[i] = list(env.step(data[i]))
+                    if buf[i][2]:  # check if done
+                        buf[i][0] = env.reset()
+                remote.send(buf)
             elif cmd == 'reset':
-                ob = env.reset()
-                remote.send(ob)
+                for i, env in enumerate(envs):
+                    buf[i] = env.reset()
+                remote.send(buf)
             elif cmd == 'render':
-                remote.send(env.render(mode='rgb_array'))
+                for i, env in enumerate(envs):
+                    buf[i] = env.render(mode='rgb_array')
+                remote.send(buf)
             elif cmd == 'close':
                 remote.close()
                 break
             elif cmd == 'get_spaces_spec':
-                remote.send((env.observation_space, env.action_space, env.spec))
+                remote.send((envs[0].observation_space, envs[0].action_space, envs[0].spec))
```

Comment:
should this send the specs for all envs (rather than only the spec for the first env) ?

Reply:
oh nvm looks like we are only using the spec from the very first env anyways